components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -11786,4 +11786,35 @@ public void deleteApplicationAttributes(String attributeKey, int applicationId)
             APIMgtDBUtil.closeAllConnections(ps, connection, null);
         }
     }
+
+    /**
+     * Add new attributes against an Application in API Store
+     *
+     * @param applicationAttributes Map of key, value pair of attributes
+     * @param applicationId
+     * @param tenantId
+     * @throws APIManagementException
+     */
+    public void addApplicationAttributes(Map<String, String> applicationAttributes, int applicationId, int tenantId)
+            throws APIManagementException {
+
+        Connection connection = null;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            connection.setAutoCommit(false);
+            addApplicationAttributes(connection, applicationAttributes, applicationId, tenantId);
+            connection.commit();
+        } catch (SQLException sqlException) {
+            if (connection != null) {
+                try {
+                    connection.rollback();
+                } catch (SQLException e) {
+                    log.error("Failed to rollback add application attributes ", e);
+                }
+            }
+            handleException("Failed to add Application", sqlException);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(null, connection, null);
```

Comment:
Need to setAutoCommit(true) in finally block
APIMgtDBUtil.setAutoCommit(connection, initialAutoCommit);
@bhathiya WDYT?

Reply:
Since the connection gets closed in finally block, do we need to setAutoCommit back to its initial state.